controllers/storageclassrequest/storageclassrequest_controller_test.go

```diff
@@ -218,6 +267,201 @@ func TestProfileReconcile(t *testing.T) {
+						Name: "spec",
+						PoolSpec: rookCephv1.PoolSpec{
+							FailureDomain: "zone",
+							DeviceClass:   deviceClass,
+							Parameters:    map[string]string{},
+						},
+					},
+				},
+			},
+		},
+		{
+			label:            "rejected profile",
+			expectedPoolName: "test-rejected-blockpool",
+			failureExpected:  true,
+			storageProfile:   rejectedStorageProfile,
+			createObjects: []runtime.Object{
+				&rookCephv1.CephBlockPool{
+					ObjectMeta: metav1.ObjectMeta{
+						Name:      "test-rejected-blockpool",
+						Namespace: namespaceName,
+						Labels: map[string]string{
+							controllers.StorageConsumerNameLabel: fakeStorageConsumer.Name,
+							controllers.StorageProfileSpecLabel:  rejectedStorageProfile.GetSpecHash(),
+						},
+					}, Spec: rookCephv1.NamedBlockPoolSpec{
+						Name: "spec",
+						PoolSpec: rookCephv1.PoolSpec{
+							FailureDomain: "zone",
+							DeviceClass:   deviceClass,
+							Parameters:    map[string]string{},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	for _, c := range primaryTestCases {
+		caseCounter++
+		caseLabel := fmt.Sprintf("Case %d: %s", caseCounter, c.label)
+		fmt.Println(caseLabel)
+
+		r := createFakeReconciler(t)
+		if strings.Contains(c.label, "rejected") {
+			r.storageCluster.Spec.DefaultStorageProfile = rejectedStorageProfile.Name
+		}
+		r.StorageClassRequest.Spec.Type = "blockpool"
+
+		r.StorageClassRequest.Spec.StorageProfile = c.storageProfile.Name
+
+		c.createObjects = append(c.createObjects, c.storageProfile)
+		c.createObjects = append(c.createObjects, fakeStorageConsumer)
+
+		fakeClient := fake.NewClientBuilder().WithScheme(r.Scheme).WithRuntimeObjects(c.createObjects...)
+		r.Client = fakeClient.Build()
+
+		_, err = r.reconcilePhases()
+		if c.failureExpected {
+			assert.Error(t, err, caseLabel)
+			continue
+		}
+		assert.NoError(t, err, caseLabel)
+
+		assert.Equal(t, c.expectedPoolName, r.cephBlockPool.Name, caseLabel)
+
+		if strings.Contains(c.expectedPoolName, "valid") {
+			expectedStorageProfileParameters := validStorageProfile.Spec.BlockPoolConfiguration.Parameters
+			actualBlockPoolParameters := r.cephBlockPool.Spec.Parameters
+			assert.Equal(t, expectedStorageProfileParameters, actualBlockPoolParameters, caseLabel)
+		} else {
+			actualBlockPoolParameters := r.cephBlockPool.Spec.Parameters
+			assert.Equal(t, v1.StorageProfilePhaseRejected, rejectedStorageProfile.Status.Phase)
+			assert.Nil(t, actualBlockPoolParameters, caseLabel)
+		}
+	}
+
+}
+
+func TestStorageProfileCephFsSubVolGroup(t *testing.T) {
+	var err error
+	var caseCounter int
+
+	var primaryTestCases = []struct {
+		label             string
+		expectedGroupName string
+		failureExpected   bool
+		createObjects     []runtime.Object
+		cephResources     []*v1alpha1.CephResourcesSpec
+		storageProfile    *v1.StorageProfile
+		cephFs            *rookCephv1.CephFilesystem
+	}{
+		{
+			label:             "valid profile",
+			expectedGroupName: "test-subvolgroup",
```

Comment:
test-subvolgroup -> test-subvolumegroup sounds more good

Reply:
`subvolgroup` is aligned with the naming convention

Reply:
Resolving as per @rchikatw's manual confirmation